packages/dd-trace/src/appsec/rasp/stack_trace.js

```diff
@@ -0,0 +1,42 @@
+'use strict'
+
+// TODO Copied from packages/dd-trace/src/appsec/iast/path-line.js, extract to a common file
+function getCallSiteInfo () {
+  const previousPrepareStackTrace = Error.prepareStackTrace
+  const previousStackTraceLimit = Error.stackTraceLimit
+  let callsiteList
+  Error.stackTraceLimit = 100
+  try {
+    Error.prepareStackTrace = function (_, callsites) {
+      callsiteList = callsites
+    }
+    const e = new Error()
+    e.stack
+  } finally {
+    Error.prepareStackTrace = previousPrepareStackTrace
+    Error.stackTraceLimit = previousStackTraceLimit
+  }
+  return callsiteList
+}
+
+function generateStackTraceForMetaStruct (maxCallSite = 32) {
+  let callSites = getCallSiteInfo()
+  let i = 0
+  if (callSites.length > maxCallSite) {
```

Comment:
We should check `maxCallSite` is a positive integer: otherwise unexpected behaviours could occur slicing frames.